### PR TITLE
Fix run program grad node mem

### PR DESCRIPTION
**Pull request by 0x45f**
### PR types

Performance optimization
### PR changes

Others
### Description

Fix run program grad node mem
PCard-66972

### CODE REVIEW

File: paddle/fluid/eager/to_static/run_program_op_func.h [hunk 1]
```diff
@@ -53,6 +54,41 @@ static void clear_no_grad_edges_with_partial_block(
...
+      if (var->IsType<phi::DenseTensor>()) {
+        garbages->emplace_back(
+            var->GetMutable<phi::DenseTensor>()->MoveMemoryHolder());
+      }
+    }
+  }
+  delete garbages;
+}
+
+static std::vector<paddle::Tensor> filte_unused_input_var_in_backward(
```
Comment (2742195759) [inline paddle/fluid/eager/to_static/run_program_op_func.h]: filter?

Comment (0x45f) [reply paddle/fluid/eager/to_static/run_program_op_func.h]: 已修改

**Review by 2742195759 — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**